quarkus.security.jdbc.principal-query.sql is required even if quarkus.security.jdbc.enabled is set to false

`quarkus.security.jdbc.principal-query.sql` is required even if `quarkus.security.jdbc.enabled=false`

Expected behavior is to not require `quarkus.security.jdbc.principal-query.sql`  when `quarkus.security.jdbc.enabled=false` is set.

```
[INFO] Running io.quarkus.qe.MyResourceTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.306 s <<< FAILURE! - in io.quarkus.qe.MyResourceTest
[ERROR] testHelloEndpoint  Time elapsed: 0.011 s  <<< ERROR!
java.lang.RuntimeException: java.lang.RuntimeException: java.lang.RuntimeException: java.util.NoSuchElementException: Property quarkus.security.jdbc.principal-query.sql not found
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: java.util.NoSuchElementException: Property quarkus.security.jdbc.principal-query.sql not found
Caused by: java.lang.RuntimeException: java.util.NoSuchElementException: Property quarkus.security.jdbc.principal-query.sql not found
Caused by: java.util.NoSuchElementException: Property quarkus.security.jdbc.principal-query.sql not found
```

Reproducer:
```
rm -rf fooBar
mvn io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:create \
    -DprojectGroupId=io.quarkus.qe \
    -DprojectArtifactId=fooBar \
    -DprojectVersion=1.0.0-SNAPSHOT \
    -DplatformArtifactId=quarkus-bom \
    -DclassName="io.quarkus.qe.MyResource" \
    -Dextensions="quarkus-elytron-security-jdbc"

echo "quarkus.security.jdbc.enabled=false" > fooBar/src/main/resources/application.properties
mvn -f fooBar/pom.xml clean verify
```

--- a/extensions/elytron-security-jdbc/runtime/src/main/java/io/quarkus/elytron/security/jdbc/JdbcRecorder.java
+++ b/extensions/elytron-security-jdbc/runtime/src/main/java/io/quarkus/elytron/security/jdbc/JdbcRecorder.java
@@ -43,11 +43,10 @@ public class JdbcRecorder {
     }
 
     private void registerPrincipalQuery(PrincipalQueryConfig principalQuery, JdbcSecurityRealmBuilder builder) {
-        DataSource dataSource = (DataSource) principalQuery.datasource
-                .map(name -> Arc.container().instance(name).get())
-                .orElse(Arc.container().instance(DataSource.class).get());
 
-        QueryBuilder queryBuilder = builder.principalQuery(principalQuery.sql).from(dataSource);
+        QueryBuilder queryBuilder = builder.principalQuery(principalQuery.sql.orElseThrow(
+                () -> new IllegalStateException("quarkus.security.jdbc.principal-query.sql property must be set")))
+                .from(getDataSource(principalQuery));
 
         AttributeMapper[] mappers = principalQuery.attributeMappings.entrySet()
                 .stream()
@@ -61,5 +60,11 @@ public class JdbcRecorder {
         if (principalQuery.bcryptPasswordKeyMapperConfig.enabled) {
             queryBuilder.withMapper(principalQuery.bcryptPasswordKeyMapperConfig.toPasswordKeyMapper());
         }
+    }
+
+    private DataSource getDataSource(PrincipalQueryConfig principalQuery) {
+        return (DataSource) principalQuery.datasource
+                .map(name -> Arc.container().instance(name).get())
+                .orElse(Arc.container().instance(DataSource.class).get());
     }
 }

--- a/extensions/elytron-security-jdbc/runtime/src/main/java/io/quarkus/elytron/security/jdbc/PrincipalQueryConfig.java
+++ b/extensions/elytron-security-jdbc/runtime/src/main/java/io/quarkus/elytron/security/jdbc/PrincipalQueryConfig.java
@@ -16,7 +16,7 @@ public class PrincipalQueryConfig {
      * The sql query to find the password
      */
     @ConfigItem
-    public String sql;
+    public Optional<String> sql;
 
     /**
      * The data source to use

